catroid/src/main/java/org/catrobat/catroid/io/DeviceUserDataAccessor.java

```diff
@@ -83,20 +83,24 @@ public boolean readUserData(UserData userData) {
 		return true;
 	}
 
-	public void writeUserData(UserData userData) throws IOException {
+	public void writeUserData(UserData userData) {
 		synchronized (getLock()) {
-			if (!deviceFile.exists() && !deviceFile.createNewFile()) {
-				throw new IOException("Cannot create " + deviceFile.getAbsolutePath());
-			}
+			try {
+				if (!deviceFile.exists() && !deviceFile.createNewFile()) {
+					throw new IOException("Cannot create " + deviceFile.getAbsolutePath());
```

Comment:
Please, do not throw an exception and catch it within the same Method.
```suggestion
					Log.e(TAG, "Cannot create " + deviceFile.getAbsolutePath());
					return;
```